braindecode/datasets/base.py

```diff
@@ -142,6 +144,88 @@ def transform(self, value):
         self._transform = value
 
 
+class SequenceDataset(WindowsDataset):
+    """Dataset of sequences of consecutive windows.
+
+    Parameters
+    ----------
+    windows : mne.Epochs
+        Windows obtained through the application of a windower to a BaseDataset
+        (see `braindecode.datautil.windowers`).
+    description : dict | pandas.Series | None
+        Holds additional info about the windows.
+    transform : callable | None
+        On-the-fly transform applied to a window before it is returned.
+    seq_len : int
+        Number of consecutive windows in a sequence.
+    step_len : int
+        Number of windows between two consecutive sequences.
+    label_transform : callable | None
+        On-the-fly transform to apply to the sequence of labels. Useful to e.g.
+        select a single label for each sequence.
+    on_missing : 'raise' | 'warn' | 'ignore'
+        What to do if no sequences can be extracted given the number of epochs
+        and sequence parameters. Valid keys are 'raise' | 'warn' | 'ignore'.
+        Default is 'raise'. If on_missing is 'warn' it will proceed but warn,
+        if 'ignore' it will proceed silently.
+    """
+    def __init__(self, windows, description=None, transform=None, seq_len=21,
+                 step_len=1, label_transform=None, on_missing='raise'):
+        super().__init__(windows, description=description, transform=transform)
+
+        if seq_len > len(windows):
+            msg = ('Sequence length is larger than number of windows '
+                   f'({seq_len} > {len(windows)}).')
+            _on_missing(on_missing, msg)
```

Comment:
do you foresee a case where user with not use on_missing = 'raise' here?

Reply:
I was thinking if you're working with a large uncurated dataset you might want to just ignore files that are not long enough for the sequence length you're interested in. But I guess right now this would break anyway when calling __getitem__. Maybe I should just remove the on_missing logic?

Reply:
agreed in the `__getitem__` way it would crash anyways, hence `on_missing` could not be used...